db-engine/src/main/java/com/splicemachine/db/iapi/stats/ColumnStatisticsImpl.java

```diff
@@ -226,9 +231,45 @@ public long selectivity(DataValueDescriptor element) {
         if (count>0)
             return count;
         // Return Cardinality Based Estimate
-        return (long) ( ((double) quantilesSketch.getN())/thetaSketch.getEstimate()); // Should we remove frequent items?
+        if (rpv == -1)
+            rpv = getAvgRowsPerValueExcludingSkews();
+        return rpv;
+    }
+
+    private long getAvgRowsPerValueExcludingSkews() {
+        long skewCount = 0;
+        long skewNum = 0;
+        com.yahoo.sketches.frequencies.ItemsSketch.Row<DataValueDescriptor>[] items = frequenciesSketch.getFrequentItems(ErrorType.NO_FALSE_POSITIVES);
+        for (com.yahoo.sketches.frequencies.ItemsSketch.Row<DataValueDescriptor> row: items) {
+            skewCount += row.getEstimate();
+            skewNum ++;
+        }
+        long nonSkewedNum = (long)(thetaSketch.getEstimate() - skewNum);
+        if (nonSkewedNum <= 0)
+            nonSkewedNum = 1;
+        long nonSkewCount = quantilesSketch.getN() - skewCount;
+        if (nonSkewCount < 0)
+            nonSkewCount = 0;
+        return (long) (((double)nonSkewCount)/nonSkewedNum);
     }
 
+    private long getSkewedRowCountInRange(DataValueDescriptor start, DataValueDescriptor stop, boolean includeStart, boolean includeStop) {
+        long skewCount = 0;
+        com.yahoo.sketches.frequencies.ItemsSketch.Row<DataValueDescriptor>[] items = frequenciesSketch.getFrequentItems(ErrorType.NO_FALSE_POSITIVES);
+        for (com.yahoo.sketches.frequencies.ItemsSketch.Row<DataValueDescriptor> row: items) {
+            DataValueDescriptor skewedValue = row.getItem();
+            try {
+                if (skewedValue != null &&
+                        (start == null || start.isNull() || skewedValue.compare(includeStart ? ORDER_OP_GREATEROREQUALS : ORDER_OP_GREATERTHAN, start, false, false)) &&
+                        (stop == null || stop.isNull() || skewedValue.compare(includeStop ? ORDER_OP_LESSOREQUALS : ORDER_OP_LESSTHAN, stop, false, false)))
+                    skewCount += row.getEstimate();
+            } catch (StandardException e) {
+                // cost estimation error does not need to fail the query
```

Comment:
Please log the error here

Reply:
Thanks @dgomezferro I added the code to log the error in the splice.log if debug log is turned on.

Reply:
If we don't expect this exception to happen I would log it at WARN level, making it clear that something unexpected happened even though we don't fail the operation. 
If we expect this to happen often please add a comment explaining when that is.
And add the exception to the call so that we get the stacktrace on the log: LOG.level("msg", e);